sdk/core/azure-core/inc/azure/core/url.hpp

```diff
@@ -58,6 +58,14 @@ namespace Azure { namespace Core {
 
     std::string GetUrlWithoutQuery(bool relative) const;
 
+    /**
+     * @brief Finds the first '?' symbol and parses everything after it as query parameters.
+     * separated by '&'.
+     *
+     * @param encodedQueryParameters String containing one or more query parameters.
+     */
+    void AppendQueryParameters(const std::string& encodedQueryParameters);
```

Comment:
Interestingly, this API doesn't append the query parameters at the end, but rather in the beginning. I would have expected it to append at the end.

For example, it goes from this:
```text
https://url/path?key1=value1
```
To:
```
https://url/path?se=date&sig=signature&sp=...&spr=...&sr=bv&st=anotherdate&sv=anothervalue&key1=value1
```

@vhvb1989, @Jinming-Hu - Is it generally OK to change the order of query parameters?

Reply:
Should it be named `AddQueryParameters()` then? Prepending does not sound like "Append".
And for parameters order, I think it orders them alphabetically ignoring case? Anyways, it is Add() and not Append() if it does not puts them to the end.

Reply:
Since it is private now, the name is not as important anymore. That said I would be OK with renaming it to `AddQueryParameters`.

Reply:
It's a `std::map`. Query parameters will be in alphabetical order, right?